lib/streamaggr/streamaggr.go

```diff
@@ -372,9 +372,9 @@ func newAggregator(cfg *Config, pushFunc PushFunc, dedupInterval time.Duration)
 	}
 	suffix += "_"
 
-	var dedupAggr *lastAggrState
+	var dedup *deduplicator
 	if dedupInterval > 0 {
-		dedupAggr = newLastAggrState()
+		dedup = newDeduplicator(dedupInterval)
```

Comment:
Would it make sense to have a global deduplicator object, so we could use a global sync.Map under the hood?
This would reduce memory usage when there are more than one aggregator configured.
Or is it intended to maintain a map per-aggregator?

Reply:
Btw, if we have one global map for deduplication, can we get rid of strings interning?